R/recode2NA.R

```diff
@@ -40,15 +40,18 @@ recode2NA <- function(GADSdat, recodeVars = namesGADS(GADSdat), value = "") {
 recode2NA.GADSdat <- function(GADSdat, recodeVars = namesGADS(GADSdat), value = "") {
   check_GADSdat(GADSdat)
   if(!is.character(recodeVars) || length(recodeVars) < 1) stop("'recodeVars' needs to be character vector of at least length 1.")
-  if(!all(recodeVars %in% namesGADS(GADSdat))) stop("All variables names in 'recodeVars' need to be variables in the GADSdat.")
-  if(!is.vector(value) || length(value) != 1) stop("'value' needs to be a vector of exactly length 1.")
+  check_vars_in_GADSdat(GADSdat, vars = recodeVars)
+  if(!is.vector(value) || length(value) == 0) stop("'value' needs to be a vector of at least length 1.")
 
-  if(length(which(GADSdat$labels[GADSdat$labels$varName %in% recodeVars, "value"] == value)) > 0) {
-   # stop("'string' is labeled in at least one of the recodeVars.")
+  labeled_values <- GADSdat$labels[GADSdat$labels$varName %in% recodeVars, c("varName", "value")]
+  labeled_values_recode <- unique(labeled_values[which(labeled_values$value %in% value), "varName"])
+  if(length(labeled_values_recode) > 0) {
+   warning("Some 'value' is labeled in the following variables in 'recodeVars': ",
+        paste(labeled_values_recode, collapse = ", "))
   }

```

Comment:
Nit: Maybe add: "Please check if this is correct"? 🤔 Or maybe "Some 'value' has a 'valLabel' in 'recodeVars': " ?